### PR TITLE
Art

### PR DESCRIPTION
**Pull request by 0xCooki**
Still required before deployment:
- Art components
- Art descriptions (both individual components and overall collection description)
- Updated deployment script given additional art components

### CODE REVIEW

File: src/BBitsEmoji.sol [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    function willMintSettleRaffle() public view returns (bool) {
+        return !(block.timestamp - raffleInfo[currentDay].start < mintDuration);
+    }
+
+    /// INTERNAL ///
+
+    /// @dev User canan mint more than once per day, but their raffle entry is not updated.
+    ///      This keeps the raffle logic simpler.
+    function _mintEntry() internal {
+        _mint(msg.sender, currentDay, 1, "");
```
[the remark this answers is not in the document]
Comment (andreitr) [reply src/BBitsEmoji.sol:134]: Or people can simply click the mint button several times?

File: src/BBitsEmoji.sol [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    function userMintPrice(address _user) public view returns (uint256) {
+        (, uint16 streak,) = checkIn.checkIns(_user);
+        if (streak > 90) streak = 90;
+        return mintPrice - ((mintPrice * streak) / 100);
+    }
+
+    /// @notice This view function returns the entry struct that corresponds to the day and index pair.
+    /// @param  _day The raffle day (token Id).
+    /// @param  _index The user index in the raffleInfo array of raffle entries.
+    function userEntry(uint256 _day, uint256 _index) public view returns (Entry memory) {
```
Comment (andreitr) [inline src/BBitsEmoji.sol:118]: Can we rename this method to `useEntryByIdex` and add another method to return entry by address `userEntry(uint256 _day, address _user)` ? 

This way I will be able to display user's info in the UI without the need to loop through all the entries.

**Review by andreitr — CHANGES_REQUESTED** [review]

File: src/BBitsEmoji.sol [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        uint256 burned = burnAmount();
+        uint256 reward = raffleAmount();
+        address winner = _settle();
+        if (burned > 0) burner.burn{value: burned}(0);
+        (bool success,) = winner.call{value: reward}("");
+        if (!success) revert TransferFailed();
+        raffleInfo[currentDay].rewards = reward;
+        raffleInfo[currentDay].burned = burned;
+        raffleInfo[currentDay].winner = winner; 
+        emit Raffle(currentDay, raffleInfo[currentDay].mints, winner, reward, burned);
```
Comment (andreitr) [inline src/BBitsEmoji.sol:158]: rename `Raffle` event to `End`

File: src/BBitsEmoji.sol [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        /// Start new raffle
+        ++currentDay;
+        _set(currentDay);
+        raffleInfo[currentDay].tokenId = currentDay;
+        raffleInfo[currentDay].start = block.timestamp;
```
Comment (andreitr) [inline src/BBitsEmoji.sol:163]: Emit `Start(currentDay)` event

**Review by andreitr — CHANGES_REQUESTED** [review]

File: src/BBitsEmoji.sol [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    /// @notice Percentage of funds used to buy and burn the BBITS token.
+    /// @dev    10_000 = 100%
+    uint256 public immutable burnPercentage;
+
+    /// @notice Length of time that a raffle lasts.
+    uint256 public immutable mintDuration;
+
+    /// @notice The price to mint an NFT.
+    uint256 public mintPrice;
+
```
[the remark this answers is not in the document]
Comment (0xCooki) [reply src/BBitsEmoji.sol:35]: mint price or duration?

File: src/BBitsEmoji.sol [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        totalEntries[currentDay] = 1;
+        _pause();
+    }
+
+    receive() external payable {}
+
+    /// @notice This function provides the core functionality for this contract, including minting, creating
+    ///         raffles, settling raffles, and generating new daily art.
+    function mint() external payable nonReentrant whenNotPaused {
+        if (willMintSettleRaffle()) {
```
[the remark this answers is not in the document]
Comment (0xCooki) [reply src/BBitsEmoji.sol:66]: That is how it currently works

File: src/BBitsEmoji.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xCooki) [reply src/BBitsEmoji.sol:134]: Yep, you can mint as many times as you want per day

File: src/BBitsEmoji.sol (hunk 6 above)
[the remark this answers is not in the document]
Comment (andreitr) [reply src/BBitsEmoji.sol:70]: Oh, I see. So in order to start the new mint a user just need to call the mint function without sending any ETH?

[the remark this answers is not in the document]
Comment (0xCooki) [reply src/BBitsEmoji.sol:70]: Yep, exactly, and there's a view function `willMintSettleRaffle()` that can tell them ahead of time whether them calling mint will end the raffle and start a new one

**Pull request merged**